pep-513.rst

```diff
@@ -216,6 +211,79 @@ the list above to remove libraries that have turned out to be
 problematic or add libraries that have turned out to be safe.
 
 
+libpythonX.Y.so.1
+-----------------
+
+Note that ``libpythonX.Y.so.1`` is *not* on the list of libraries that
+a ``manylinux1`` extension is allowed to link to. It turns out that
+this is unnecessary in almost all cases: the way ELF linking works,
+extension modules that are loaded into the interpreter automatically
+get access to all of the interpreter's symbols, regardless of whether
+or not the extension itself is explicitly linked against
+libpython. Furthermore, explicit linking to libpython creates problems
+in the common configuration where Python is not built with
+``--enable-shared``. In particular, on Debian and Ubuntu systems,
+``apt install pythonX.Y`` does not even install ``libpythonX.Y.so.1``,
+meaning that any wheel that *did* depend on ``libpythonX.Y.so.1``
+could fail to import.
+
+There is one situation where extensions that are linked in this way
+can fail to work: if a host program (e.g., ``apache2``) uses
+``dlopen()`` to load a module (e.g., ``mod_wsgi``) that embeds the
+CPython interpreter, and the host program does *not* set the
+``RTLD_GLOBAL`` flag, then the embedded CPython will be unable to load
+any extension modules that do not themselves link explicitly to
+``libpythonX.Y.so.1``. Fortunately, ``apache2`` *does* set the
+``RTLD_GLOBAL`` flag, as do all the other programs that
+embed-CPython-via-a-dlopened-plugin that we could locate, so this does
+not seem to be a serious problem in practice. The incompatibility with
+Debian/Ubuntu is more of an issue than the theoretical incompatibility
+with a rather obscure corner case.
+
+This is a rather complex and subtle issue that extends beyond
+the scope of ``manylinux1``; for more discussion see: [9]_, [10]_,
+[11]_.
+
+UCS-2 vs UCS-4 builds
+---------------------
+
+All versions of CPython 2.x, plus CPython 3.0-3.2, can be built in two
+ABI-incompatible modes: one where Unicode is stored using UCS-2, and
+one where it's stored using UCS-4 (controlled by
+``--enable-unicode=ucs2`` versus ``--enable-unicode=ucs4``; CPython
+3.3 and greater use a different storage method that always supports
+UCS-4). If we want to make sure UCS-2 wheels don't get installed into
+UCS-4 CPythons and vice-versa, then something must be done.
+
+An earlier version of this PEP included a requirement that manylinux1
+wheels targeting these older CPython versions should always use the
+UCS-4 ABI. But then, in between the PEP's initial acceptance and its
+implementation, ``pip`` and ``wheel`` gained first-class support for
+tracking and checking this aspect of ABI compatibility for the
+relevant CPython versions, which is a better solution. So we now allow
```

Comment:
PEP's normally spend a lot of time talking about other things that were considered but rejected, etc.; given that we're actually changing the PEP after it was accepted I think it's a good idea to record the reasoning.